clients/e-shop-demo/e-shop-frontend/src/components/checkout-iota/checkout-iota.component.tsx

```diff
@@ -0,0 +1,68 @@
+import { useContext, useRef, useState } from 'react';
+import { CheckoutStepHeading, CheckoutWithIotaContainer, CheckoutWithIotaContainerHeading } from './checkout-iota.styles';
+import { verifyCredential, readJSON, checkAge } from '../../services/verify-credential.service';
+import { UserContext } from '../../contexts/user.provider';
+import GenerateNonce from '../generate-nonce/generate-nonce.component';
+import AuthorizeCredential from '../authorize-credential/authorize-credential.component';
+import { SmallButton } from '../../global.styles';
+import MessageBox from '../message-box/message-box.component';
+
+const CheckoutWithIota = () => {
+	const inputRef = useRef<any>();
+	const [aboveAgeRestriction, setAboveAgeRestriction] = useState<boolean>();
+	const [file, setFile] = useState<File>();
+	const { setCredential, authenticated, isVerified, setIsVerified } = useContext(UserContext);
+
+	const onVerify = async () => {
+		setIsVerified(false);
+		const credential = await readJSON(file as File);
+		const verified = await verifyCredential(credential);
+		const ageCheck = checkAge(credential);
+		setAboveAgeRestriction(ageCheck);
+		if (verified && ageCheck) {
+			setIsVerified(verified);
+			setCredential(credential);
```

Comment:
setIsVerified should be called everytime and then line 17 can be removed. imo credential could also always be set